psl/builtin-connectors/src/mysql_datamodel_connector.rs

```diff
@@ -237,6 +237,20 @@ impl Connector for MySqlDatamodelConnector {
         }
     }
 
+    fn validate_datasource(
+        &self,
+        preview_features: BitFlags<PreviewFeature>,
+        ds: &Datasource,
+        errors: &mut Diagnostics,
+    ) {
+        if ds.provider == "@prisma/mysql" && !preview_features.contains(PreviewFeature::NodeDrivers) {
+            errors.push_error(DatamodelError::new_validation_error(
+                "The provider \"@prisma/mysql\" is currently in preview and must be explicitly enabled. Please add the preview feature flag \"nodeDrivers\" to the generator block in your schema.prisma file.",
+                Span::empty(),
```

Comment:
should we maybe add `provider_span` to `Datasource` and use it here?

Reply:
Hi @aqrln, how so?